dracut/30ignition/ignition-prepare.target

```diff
@@ -0,0 +1,28 @@
+# This target is reached when Ignition is done being prepeared. This
+# activated *only* on first boot (or if ignition.firstboot=1 is provided).
+# Thus, it is also an API for units to use so that they are activated only on
+# first boot. Simply add a link under ignition-prepare.target.requires in the
+# initrd.
+
+[Unit]
+Description=Ignition Prepare
+Documentation=https://github.com/coreos/ignition
+ConditionPathExists=/etc/initrd-release
+Before=local-fs-pre.target
+Before=initrd.target
+Before=ignition-diskful.target
+Before=ignition-complete.target
+
+# initrd.target has OnFailureJobMode=replace-irreversibly, which seems to
+# cause unit restart loops in the initramfs if one of our units fails.  Thus
+# we intercept Ignition unit failures here and isolate to emergency.target.
+# https://github.com/coreos/ignition-dracut/issues/115
+OnFailure=emergency.target
+OnFailureJobMode=isolate
+
+# Prepare and then unload.
+Conflicts=ignition-diskful.target ignition-complete.target
```

Comment:
Hmm, so the idea here is that after we pass `ignition-prepare.target`, we unmount `/boot` automatically? And we do that because otherwise `ignition-disks` for some reason fails to wait for the partitions it created?

Reply:
Correct. It was dead-tricky to debug. However, I found that the `/dev/disk/by-{label,uuid}` are not updated, which causes the CI to failure.